cmd/kubeadm/app/cmd/join.go

```diff
@@ -282,15 +283,20 @@ func NewJoin(cfgPath string, args []string, defaultcfg *kubeadmapiv1beta1.JoinCo
 		return nil, err
 	}
 
+	initcfg, err := fetchInitConfigurationFromJoinConfiguration(internalcfg)
```

Comment:
one interesting question here (to me at least), with this we are trying to fetch the init configuration, **before** we have checked if this node is healthy.
https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/preflight/checks.go#L933

is this a good idea?

Reply:
Another option would be to split the checks into two groups, we run the general join node checks; if everything is fine we then fetch the init configuration and run the checks that apply to that init configuration, kind of splitting `RunJoinNodeChecks` into `RunPreJoinNodeChecks` and `RunPostJoinNodeChecks` or something similar?